pdns/misc.cc

```diff
@@ -796,7 +796,12 @@ int makeUNsockaddr(const std::string& path, struct sockaddr_un* ret)
   if (path.length() >= sizeof(ret->sun_path))
     return -1;
 
-  path.copy(ret->sun_path, sizeof(ret->sun_path), 0);
+  if (path.find("abns@", 0, 5) == 0) {
```

Comment:
Also, `find` here does not guarantee the match will occur at the beginning of the string.
Why not simply use
```suggestion
  if (path.substr(0, 5) == "abns@") {
```